devito/passes/iet/mpi.py

```diff
@@ -122,22 +143,22 @@ def _merge_halospots(iet):
         for hs in halo_spots[1:]:
             mapper[hs] = hs.halo_scheme
 
-            for f in hs.fmapper:
+            for f, (loc_indices, _) in hs.fmapper.items():
```

Comment:
why not the same as before and just pass `hs` to rule and let it take what it needs in it.

Reply:
as explained on Slack, not sure I can drop this loop, because `f` is used below. I agree I could rewrite stuff somewhere in halo_scheme.py such that eventually this loop body becomes less contrieved (I don't like it either to pass `hs` and `loc_indices` to `rule`) , but unless you insist I'd leave it for another day?